noxfile.py

```diff
@@ -49,14 +33,55 @@
 reset_coverage = True
 
 
+def install_deps(
+    session: nox.Session,
+    *,
+    extras: Sequence[str] = (),
+    groups: Sequence[str] = (),
+    project: bool = True,
+) -> None:
+    """Helper to install dependencies from a group."""
+    if not project and extras:
+        raise TypeError("Cannot install extras without also installing the project")
+
+    command: List[str] = [
+        "pdm",
+        "sync",
+        "--fail-fast",
+        "--only-keep",
```

Comment:
```suggestion
        "--clean-unselected",
```
same result, but the flag name feels a little more descriptive